vql/server/repository.go

```diff
@@ -95,6 +95,72 @@ func (self ArtifactSetFunction) Info(
 	}
 }
 
+type ArtifactDeleteFunctionArgs struct {
+	Name string `vfilter:"optional,field=name,doc=The Artifact to delete"`
+}
+
+type ArtifactDeleteFunction struct{}
+
+func (self *ArtifactDeleteFunction) Call(ctx context.Context,
+	scope *vfilter.Scope,
+	args *ordereddict.Dict) vfilter.Any {
+
+	arg := &ArtifactDeleteFunctionArgs{}
+	err := vfilter.ExtractArgs(scope, args, arg)
+	if err != nil {
+		scope.Log("artifact_delete: %v", err)
+		return vfilter.Null{}
+	}
+
```

Comment:
We need a permissions check here

Reply:
Added the same switch statement from artifact_set.